docs/_en/dev/feature-technical-designs/1696-gcp-secrets.md

```diff
@@ -0,0 +1,277 @@
+---
+title: GCP - Overall secret management design
+---
+
+# GCP - Overall secret management design #1696
+
+- Proposal PR: [https://github.com/theopenconversationkit/tock/pull/1697](https://github.com/theopenconversationkit/tock/pull/1697)
+- Github Issue for this feature: https://github.com/theopenconversationkit/tock/issues/1696
+
+
+API version used : v1 of Google Secret Manager, reference version here.
+
+## Introduction
+
+We have already an integration of AWS Secret manager essentially used for :
+* **Generative AI services** dynamic secrets management, secrets vault are create dynamically for generative AI services (LLM, Embedding, Observability, VectorDB) 
+* **MongoDB credentials** : Add an implementation of `ai.tock.shared.security.mongo.MongoCredentialsProvider` ([MongoCredentialsProvider](https://github.com/theopenconversationkit/tock/blob/tock-24.3.4/shared/src/main/kotlin/security/mongo/MongoCredentialsProvider.kt#L24)) for GCP. Currently we only have the connexion string (thought environment variables), but credentials can be provided using ([MongoCredentialsProvider](https://github.com/theopenconversationkit/tock/blob/tock-24.3.4/shared/src/main/kotlin/security/mongo/MongoCredentialsProvider.kt#L24)) (see [MongoClient creation here](https://github.com/theopenconversationkit/tock/blob/tock-24.3.4/shared/src/main/kotlin/Mongos.kt#L178), [asyncMongoClient here](https://github.com/theopenconversationkit/tock/blob/tock-24.3.4/shared/src/main/kotlin/Mongos.kt#L197))
+* **iAdvize Connector GraphQL credentials** : currently we have an AWS Secret Manager implementation of [`ai.tock.shared.security.credentials.CredentialsProvider`](https://github.com/theopenconversationkit/tock/blob/tock-24.3.4/shared/src/main/kotlin/security/credentials/CredentialsProvider.kt#L19) this will need to be implemented also for GCP Secret Manager. 
+
+Each of these 3 differents secret usages can use different providers idependently, for instance you could provide your mongodb credential using environment variables and handle the generate ai related secrets using GCP Secret Manager.
```

Comment:
@AurelienSylvan we have decided with @assouktim that secret providers should be independant for each scopes : gen-ai secrets, database (mongodb) secrets, iadvize secrets

In our case we will have : 
- gen ai secrets using GCP Secret Manager
- mongodb secrets / credentials using environment variable (kube secrets behind it)
- iAdvize secret using either environment variable (kube secret, we need to create it) or GCP secret ? ==> We need to decide for that point this will have role attribution impact if we use GCP Secret (see section [Role `tock.iadvize-secret.shareholder`](https://github.com/theopenconversationkit/tock/pull/1697/files#diff-846320cbce302e5bc02e2637c1343860db322ee9127551bd4ae715c10cd0d9ecR126) )

Reply:
We will use Kubescret (ENV) for iAdvize also.